### PR TITLE
docs: update limit count plugin doc

### PR DESCRIPTION
**Pull request by monkeyDluffy6017**
### Description

update limit count plugin doc

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Review by hf400159 — APPROVED** [review]

File: docs/en/latest/plugins/limit-count.md [hunk 1]
```diff
@@ -181,7 +181,13 @@ curl -i http://127.0.0.1:9180/apisix/admin/services/1 \
 }'
 ```
 
-Now every request will share the same count limitation regardless of their remote address.
+The above configuration means that when the `group` attribute of the `limit-count` plugin is configured to `services_1#1640140620` for multiple routes, requests to those routes will share the same counter, even if the requests come from different IP addresses.
+
+:::note notice
```
Comment (hf400159) [inline docs/en/latest/plugins/limit-count.md:186]:
```suggestion
:::note
```

Comment (hf400159) [reply docs/en/latest/plugins/limit-count.md:186]: English does not need to be annotated.

**Review by SylviaBABY — APPROVED** [review]

File: docs/en/latest/plugins/limit-count.md (hunk 1 above)
Comment (monkeyDluffy6017) [reply docs/en/latest/plugins/limit-count.md]: Done

**Review by SylviaBABY — APPROVED** [review]

**Review by hf400159 — APPROVED** [review]

File: docs/en/latest/plugins/limit-count.md [hunk 2]
```diff
@@ -181,7 +181,13 @@ curl -i http://127.0.0.1:9180/apisix/admin/services/1 \
 }'
 ```
 
-Now every request will share the same count limitation regardless of their remote address.
+The above configuration means that when the `group` attribute of the `limit-count` plugin is configured to `services_1#1640140620` for multiple routes, requests to those routes will share the same counter, even if the requests come from different IP addresses.
```
Comment (juzhiyuan) [inline docs/en/latest/plugins/limit-count.md:184]: What's `services_1#1640140620`?

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**